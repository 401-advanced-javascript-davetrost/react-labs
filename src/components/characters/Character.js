import React from 'react';
import propTypes from 'prop-types';
import styles from './Character.css';

const Character = ({ characters }) => {
  return (
    <>
      {characters.map(character => {
        const species = character.species || 'unknown origin';
        return (
          <div className={styles.Character + ' col s6 m4'} key={character.img}>
            <div className="card">
              <div className="card-image">
                <img src={character.img}/>
                <span className="card-title">{character.name}</span>
              </div>
              <div className="card-content">
                <p>Species: <strong>{species}</strong></p>
              </div>
            </div>
          </div>
        );
      })}
    </>
  );
};

Character.propTypes = {
  characters: propTypes.array.isRequired,
};

export default Character;
