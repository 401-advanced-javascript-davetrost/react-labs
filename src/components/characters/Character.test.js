import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character character={{  img: 'https://rickandmortyapi.com/api/character/avatar/327.jpeg', name: 'Slow Mobius', species: 'Humanoid' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
