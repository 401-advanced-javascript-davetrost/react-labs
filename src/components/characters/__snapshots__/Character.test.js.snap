// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<div
  className="Character col s6 m4"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        src="https://rickandmortyapi.com/api/character/avatar/327.jpeg"
      />
      <span
        className="card-title"
      >
        Slow Mobius
      </span>
    </div>
    <div
      className="card-content"
    >
      <p>
        Species: 
        <strong>
          Humanoid
        </strong>
      </p>
    </div>
  </div>
</div>
`;
